ntbs-service-unit-tests/Pages/HomePageTests.cs

```diff
@@ -0,0 +1,244 @@
+
+        private readonly HomepageKpi mockHomepageKpiWithPhec = new HomepageKpi
+        {
+            Code = "PHEC001",
+            PercentPositive = 17,
+            PercentHivOffered = 25,
+            PercentResistant = 11,
+            PercentTreatmentDelay = 14
+        };
+
+        private readonly HomepageKpi mockHomepageKpiWithTbService = new HomepageKpi
+        {
+            Code = "TB001",
+            PercentPositive = 15,
+            PercentHivOffered = 27,
+            PercentResistant = 12,
+            PercentTreatmentDelay = 13
+        };
+
+        public HomePageTests(HomePageFixture homePageFixture)
+        {
+            _mockAuthorizationService = new Mock<IAuthorizationService>();
+            _mockUserService = new Mock<IUserService>();
+            _mockAlertRepository = new Mock<IAlertRepository>();
+            _mockHomepageKpiService = new Mock<IHomepageKpiService>();
+            SetUpMocks();
+
+            _homePageFixture = homePageFixture;
+            _notificationRepository = new NotificationRepository(homePageFixture.Context);
+        }
+
+        private void SetUpMocks()
+        {
+            _mockAuthorizationService
+                .Setup(s => s.FilterNotificationsByUserAsync(It.IsAny<ClaimsPrincipal>(), It.IsAny<IQueryable<Notification>>()))
+                .Returns((ClaimsPrincipal user, IQueryable<Notification> notifications) => Task.FromResult(notifications));
+            _mockAuthorizationService
+                .Setup(s => s.FilterNotificationsByUserAsync(It.IsAny<ClaimsPrincipal>(), It.IsAny<IQueryable<Notification>>()))
+                .Returns((ClaimsPrincipal user, IQueryable<Notification> notifications) => Task.FromResult(notifications));
+            _mockAuthorizationService
+                .Setup(s => s.FilterAlertsForUserAsync(It.IsAny<ClaimsPrincipal>(), It.IsAny<IList<AlertWithTbServiceForDisplay>>()))
+                .Returns((ClaimsPrincipal user, IList<AlertWithTbServiceForDisplay> alerts) => Task.FromResult(alerts));
+
+            _mockUserService.Setup(s => s.GetUserType(It.IsAny<ClaimsPrincipal>())).Returns(UserType.NationalTeam);
+            _mockUserService.Setup(s => s.GetPhecCodesAsync(It.IsAny<ClaimsPrincipal>()))
+                .Returns(Task.FromResult(new List<string> { mockHomepageKpiWithPhec.Code } as IEnumerable<string>));
+            _mockHomepageKpiService.Setup(s => s.GetKpiForPhec(new List<string> { mockHomepageKpiWithPhec.Code }))
+                .Returns(Task.FromResult(new List<HomepageKpi> { mockHomepageKpiWithPhec } as IEnumerable<HomepageKpi>));
+            _mockHomepageKpiService.Setup(s => s.GetKpiForTbService(new List<string> { mockHomepageKpiWithTbService.Code }))
+                .Returns(Task.FromResult(new List<HomepageKpi> { mockHomepageKpiWithTbService } as IEnumerable<HomepageKpi>));
+        }
+
+        [Fact]
+        public async Task OnGetAsync_PopulatesPageModel_WithRecentNotifications()
```

Comment:
To guarantee that this test will pick up the behaviour change that you've implemented as part of this ticket you could add a SubmissionDate to these notifications with a different ordering (or maybe better to have a separate test, as this one is testing the case when SubmissionDate is null).

Reply:
I've added a separate test